src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -787,6 +780,37 @@ protected override void EndProcessing()
                 ThrowTerminatingError(errorRecord);
                 return;
             }
+
+            PopulateSource();
+        }
+
+        // We only keep the code for backward compatibility.
+        internal void PopulateSource()
+        {
+            // Prevent code compilation in ConstrainedLanguage mode
+            if (SessionState.LanguageMode == PSLanguageMode.ConstrainedLanguage)
+            {
+                ThrowTerminatingError(
+                    new ErrorRecord(
+                        new PSNotSupportedException(AddTypeStrings.CannotDefineNewType), "CannotDefineNewType", ErrorCategory.PermissionDenied, null));
+            }
+
+            // Load the source if they want to load from a file
+            if (String.Equals(ParameterSetName, "FromPath", StringComparison.OrdinalIgnoreCase) ||
+                String.Equals(ParameterSetName, "FromLiteralPath", StringComparison.OrdinalIgnoreCase)
+                )
+            {
+                sourceCode = "";
```

Comment:
Don't we want to use the updated StringBuilderCode here to get the perf benefit?  Also there is duplicate code that could be handled with base class helper methods.  But I believe you are looking to refactor this code so it may not be worth changing now just to be refactored later.

Reply:
I believe the method is a candidate to remove. I've tried add [Obsolete] but it don't work. I think we could document the method and perhaps  the class as deprecated - I'll wonder if anybody use it. Another argument for removing this code is that the code is the remainder of the CodeDom.
So I split the code (no helper methods) to easily remove the deprecated code later.

Reply:
Ok.

Reply:
I think the breaking changes to `AddTypeCommandBase` is most likely safe. I searched on Github and don't see the type `AddTypeCommandBase` gets used except AddType.
Given that, would it be OK that we just keep the breaking change made in #5243?

Reply:
I don't see the need for `AddTypeCommandBase`. If the team approves, I'll be happy to remove that.

Reply:
I'm merging this PR. We can loop in the powershell committee in your subsequent add-type refactoring PR (or the issue for that work).

Reply:
The request is in #5158.